sale_layout_category_hide_detail/static/src/js/boolean_fa_icon_widget.js

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2019 Tecnativa - Ernesto Tejeda
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+ */
+odoo.define('sale_layout_category_hide_detail.boolean_fa_icon_widget', function (require) {
+    "use strict";
+
+    var AbstractField = require('web.AbstractField');
+    var registry = require('web.field_registry');
+
+    var BooleanFaIconWidget = AbstractField.extend({
```

Comment:
Please check some comments below on a consideration if this is really needed.

If we see it's really needed, then:

1. Here it would make more sense to inherit from `FieldBoolean`, [just as `boolean_toggle` widget does to simplify its code](https://github.com/odoo/odoo/blob/674ee035ca954a57341cbaa951d2f6f31b5f8fe2/addons/web/static/src/js/fields/basic_fields.js#L2143-L2164).
1. The widget should go in a separate module in OCA/web IMHO.

Reply:
- boolean_toggle is a FieldBoolean with the checkbox made up with scss, but this new widget is not rendered as a checkbox and its behavior is not the same as the FieldBoolean. If I inherited from a FieldBoolean I would have to rewrite almost all FieldBoolean methods to disable its behavior, and then the inheritance would no longer make sense. It is the same as what happens with FieldBooleanButton, so that inherits from AbstractField instead of FieldBoolean
- regarding to have the widget in a separated module I agree with Pedro (https://github.com/OCA/sale-reporting/pull/70#discussion_r301590394)